### PR TITLE
Allow for domain-specific studies to be included/excluded

### PR DESCRIPTION
**Pull request by jesusff**
Copying here a request from S. Somot:

> I would like to implement for the MED GCM selection procedure. With respect to the current situation, I would like remove one study, add some new ones and change some of the existing. I think that CMIP6_studies_config.yaml is the right file to use but I 
don't know how to code it in the file.
>
> For example, if I'd like to remove `Dobler SST rmse EUR` and add
>
> `Atlas Dtas MED DJF` for the ssp370
> `Atlas Dtas MED JJA` for the ssp370
> `Atlas Dpr MED DJF` for the ssp370
> `Atlas Dpr MED JJA` for the ssp370
>
> Being able to select one by one the "key" of the selected criteria would be perfect. In this case, using only the spatial scope is not enough.

### CODE REVIEW

**Comment by jesusff** [comment]
This is not implemented, but I see several options. The simplest for the 
moment would be to open a branch for the MED files so you can disable 
there the columns that you like, with the standard procedure. For 
example, there are several entries disabled here: [AtlasIPCC.yaml](https://github.com/WCRP-CORDEX/cmip6-for-cordex/blob/5da0e4c5d/CMIP6_studies/AtlasIPCC.yaml)

The problem with changing this in the main branch is that the entry 
would be disabled for all domains. Thus, the use of the separate branch, 
where I could produce the tables for MED and feed them back to the main 
website.

In the long term, this is not ideal. The option you propose, i.e. to 
have entry like

```yaml
disabled_by_domain:
   MED:
     - Dobler SST rmse EUR
     - Atlas Dtas MED DJF
     - Atlas Dtas MED JJA
     - Atlas Dpr MED DJF
     - Atlas Dpr MED JJA
```

in CMIP6_studies_config.yaml would be one option. But this would be hard 
to maintain if the "keys" change and there is no track of why they were 
disabled.

Another way would be to have a e.g. disabled_MED field in the study yaml 
file itself. In this way, it can be explained there why it is disabled 
and other domains can consider also to disable it (maybe finally leading 
to a "disabled" field which would remove the entry for all domains). I 
mean something like this:
```yaml
disabled_MED:
     cause: not_relevant
     comment: >
       This metric is not relevant for Med-CORDEX because ...
```
what do you think? I can open the branch for MED anyway and we can start 
with the fast approach and develop there the implementation of this last 
one. When this is ready, we can merge it back to the main branch for 
other domains to benefit as well.

**Comment by jesusff** [comment]
> ok for implementing the fast approach that is to say opening a new branch for MED in which I will be able to modify all the yaml file to activate or deactivate the metrics I would like.
> On the long term, I would prefer to use the CMIP6_studies_config.yaml as the main driving file, and not to "disabled" some metrics as you proposed but more to activate them. So the users will do a positive action in selecting the metrics he/she wants to implement for a given domain. It will be easier to read and understand as we will have the list of activated metrics in `CMIP6_studies_config.yaml`. The problem of "changing keys" for a given yaml file is indeed an issue

OK, the new branch is ready https://github.com/WCRP-CORDEX/cmip6-for-cordex/tree/MED

I realized that the problem that you posted is bit more complex than I initially thought, since you plan to activate particular scenarios of a given score. Maybe it is advisable to use the ssp370 also in EURO-CORDEX, since this is the first scenario to consider.

On the other hand, the spatial scope filter was devised to select automatically all scores relevant for a given domain. These are expected to be too many, so the user should be essentially removing some of them to make the table more manageable. When new metrics are added, they would appear on the table, so the user might need to decide whether to keep them or not. Otherwise, new metrics would go unnoticed.

Anyway, lets keep thinking on solutions tailored for the different domains. Both options should be easily implemented.

File: CMIP6_studies_config.yaml
Comment (sam-somot) [inline CMIP6_studies_config.yaml:1]: I have the feeling that MED+CEU+NEU = EUR. Do we merge ?

File: docs/CMIP6_for_CORDEX_availability_ESGF.csv
Comment (sam-somot) [inline docs/CMIP6_for_CORDEX_availability_ESGF.csv:1]: concerning the availability, would it be possible to switch to OK if only HIST, SSP126 and SSP370 are available. That would be the minimum availability criteria for Med-CORDEX whereas EURO-CDX asks for 4 SSPs. So even if a GCM does not have SSP585, we could keep it in MED.

Comment (sam-somot) [inline docs/CMIP6_for_CORDEX_availability_ESGF.csv:1]: For coupled Med-CORDEX runs, we need to have monthly mean files for 3D T, 3D S and sea level available from the driving GCMs. Would it be possible to add this in the availability criteria. I do expect those fields to be available for all GCMs as we ony need monthly-mean data

**Review by sam-somot — APPROVED** [review]

**Comment by sam-somot** [comment]
I'm still quite  new to github. I tried to approve the review without making mistakes in order to allow you to merge. Very good if we can have only one branch to maintain. The disabled_DOMAIN seems a good solution

**Comment by jesusff** [comment]
Thank you, @sam-somot 

This was also a test to see if you could review pull requests and it worked fine. The review process will be simpler when there are just additions or small modifications to yaml files. In this case there were quite a few changes to the code as well. I will merge now and open specific issues with some questions you posted above.

File: docs/CMIP6_for_CORDEX_availability_ESGF.csv
Comment (jesusff) [reply docs/CMIP6_for_CORDEX_availability_ESGF.csv:1]: Moved to new issue #12

Comment (jesusff) [reply docs/CMIP6_for_CORDEX_availability_ESGF.csv:1]: Yes, this is a duplicate of #9

File: CMIP6_studies_config.yaml
Comment (jesusff) [reply CMIP6_studies_config.yaml:1]: moved to new issue #13

**Pull request merged**